### PR TITLE
enhance(pods): configure export pod v2  to create new service and pod configs.

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR adds an option to create a new service connection and export pod v2 config. Apart from this, it adds a few fixes that were found relevant while testing export pod v2 in general.

---
madge report:
- master: 65 circular dependencies
- this branch: 65 circular dependencies
---

## Testing
- [x] [Write Tests](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#writing-tests) 
- [x] [Confirm existing tests pass](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#executing-tests)
- [x] [Confirm manual testing](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#manual-testing) 
- [ ] If your tests changes an existing snapshot, make sure that snapshots are [updated](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#updating-test-snapshots)
- [ ] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in the [test workspace](https://docs.dendron.so/notes/dtMsF12SF2SUhLN10sYe2.html)
### Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://docs.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### CODE REVIEW

File: packages/plugin-core/src/commands/pods/ConfigureServiceConnection.ts [hunk 1]
```diff
@@ -12,50 +9,43 @@ import { BasicCommand } from "../base";
...
     const mngr = new ExternalConnectionManager(getExtension().podsDir);
-    const existingConnections = await mngr.getAllConfigsByType(serviceType);
-    if (existingConnections.length > 0) {
-      const items = existingConnections.map<QuickPickItem>((value) => {
-        return { label: value.connectionId };
-      });
-      const userChoice = await window.showQuickPick(items, {
+    const allServiceConfigs = await mngr.getAllValidConfigs();
+    const items = allServiceConfigs.map<QuickPickItem>((value) => {
+      return { label: value.connectionId };
```
Comment (jonathanyeung) [inline packages/plugin-core/src/commands/pods/ConfigureServiceConnection.ts:27]: Can you also add `detail: value.serviceType` in the quickpick items?

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/commands/pods/ConfigureServiceConnection.ts (hunk 1 above)
Comment (Harshita-mindfire) [reply packages/plugin-core/src/commands/pods/ConfigureServiceConnection.ts]: added service type in description.

**Comment by kevinslin** [comment]
are there docs for this change?

**Comment by jonathanyeung** [comment]
> are there docs for this change?

Part of https://github.com/dendronhq/dendron-site/pull/363/files?show-viewed-files=true&file-filters%5B%5D=

**Pull request merged**